pkg/db/transaction.go

```diff
@@ -5,30 +5,66 @@ import (
 	"fmt"
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
+	"math/rand"
 	"sync"
+	"time"
 )
 
+const txMaxRetries = 3
+const txMaxJitter = 350
+const txMinJitter = 25
+
 func TransactionResult(conn Connection, dbOps func(tx *TxConnection) (interface{}, error), logger *zap.SugaredLogger) (interface{}, error) {
+	var result interface{}
+	var txErr error
+
 	log := func(msg string, args ...interface{}) {
-		if logger != nil {
-			logger.Debugf(msg, args...)
+		if logger == nil {
+			return
 		}
+		logger.Infof(msg, args...)
 	}
-	txConnection, err := conn.Begin()
-	if err != nil {
-		return nil, err
-	}
 
-	result, err := dbOps(txConnection)
-	if err != nil {
-		log("Rollback transactional DB context because an error occurred: %s", err)
-		if rollbackErr := txConnection.Rollback(); rollbackErr != nil {
-			err = errors.Wrap(err, fmt.Sprintf("Rollback of db operations failed: %s", rollbackErr))
+	//retry the transaction until txMaxRetries is reached. Each retry has a few msec delay for better load balancing
+	for i := 0; i < txMaxRetries; i++ {
+		if i > 0 {
+			rand.Seed(time.Now().UnixNano())
+			//nolint:gosec //no security relevance, linter complains can be ignored
+			jitter := time.Duration(rand.Int63n(txMaxJitter))
```

Comment:
Isn't that Duration in range [0..350) nanoseconds?
I vote for extracting that to a separate function with an integer argument that returns a Duration.
Also add a simple unit test that verifies if the returned random Duration is within bounds.
The `time.Sleep(jitter)` should stay here and we shouldn't unit-test it's invocation.